packages/sanity/src/core/components/react-track-elements/hooks.ts

```diff
@@ -0,0 +1,116 @@
+import {debounce} from 'lodash'
+import {useLayoutEffect, useMemo, useReducer, useRef, useState} from 'react'
+
+import {type IsEqualFunction} from './types'
+
+/** @internal */
+export interface TrackerContextStore<Value> {
+  add: (id: string, value: Value) => void
+  update: (id: string, value: Value) => void
+  remove: (id: string) => void
+}
+
+function createStore<Value>(reportedValues: Map<string, Value>, publish: () => void) {
+  function add(id: string, value: Value) {
+    if (reportedValues.has(id)) {
+      // eslint-disable-next-line no-console
+      // console.error(
+      //   new Error(
+      //     `Invalid call to useReporter(${id}): A component reporting on "${id}" is already mounted in the subtree. Make sure that all reporters within the same <Tracker> subtree have unique ids.`
+      //   )
+      // )
```

Comment:
Seems like this if condition is empty (only with commented code?) I'm assuming something is meant to have gone in here?

Reply:
The same thing with the rest of the "if"s in this file 👇 
- update
- remove

Reply:
Yes, this isn't new code, it's copied from the original implementation: https://github.com/sanity-io/sanity/blob/339128dc7fda961dee8deffadd957f5fcacb1a52/packages/sanity/src/core/components/react-track-elements/createStore.ts#L15-L20

I'd like to optimize it further but that's when it started to feel like "maybe we should just use `@floating-ui/react`?" 😅 

Another point is that currently it seems like `store.add` and `store.update` does exactly the same thing. I can change things further if it helps land the PR, just wanted you to know that I stopped because it started to feel like the time would be better spent using a well maintained library off of `npm` instead of rolling our own. It seems like what we wrote 4 years ago have been largely untouched since then, and I'm probably not the only one who's a little scared when trying to change it 😬

Reply:
Yeah... I feel this as well!
I hadn't realised that the original file also had these commented. Given the size of the PR and the already commented (same) code I agree with keeping it. Probably should add a comment about floating react here as well just so that we can tackle this in the future... However if this hasn't been prioritised in 2-4 years I'm unsure if / when it will be changed 😬

Reply:
Allright, I've added a new comment block for it here: https://github.com/sanity-io/sanity/blob/589240450aecac7d84815b3954c8104363a2b0ce/packages/sanity/src/core/components/react-track-elements/hooks.ts#L14-L21 :)